velox/common/memory/HashStringAllocator.cpp

```diff
@@ -316,6 +316,46 @@ void HashStringAllocator::free(Header* _header) {
   } while (header);
 }
 
+//  static
+int64_t HashStringAllocator::offset(
+    Header* FOLLY_NONNULL header,
+    Position position) {
+  int64_t size = 0;
+  for (;;) {
+    bool continued = header->isContinued();
+    auto length = header->size() - (continued ? sizeof(void*) : 0);
+    auto begin = header->begin();
+    if (position.position >= begin && position.position <= begin + length) {
+      return size + (position.position - begin);
+    }
+    if (!continued) {
+      return -1;
+    }
+    size += length;
+    header = getNextContinued(header);
+  }
+}
+
+//  static
+HashStringAllocator::Position HashStringAllocator::seek(
+    Header* FOLLY_NONNULL header,
+    int64_t offset) {
+  int64_t size = 0;
+  for (;;) {
+    bool continued = header->isContinued();
+    auto length = header->size() - (continued ? sizeof(void*) : 0);
+    auto begin = header->begin();
+    if (offset >= size && offset < size + length) {
```

Comment:
Since `size` increases monotonically from 0, would it be enough to check only `offset < size + length` here (because if `offset > size`, `offset > size + length` should be true at the last iteration)?